util/drakeUtil.h

```diff
@@ -84,19 +93,86 @@ mxArray* stdVectorToMatlab(const std::vector<Scalar>& vec) {
   return pm;
 }
 
-DLLEXPORT const std::vector<double> matlabToStdVector(const mxArray* in);
+DLLEXPORT void sizecheck(const mxArray* mat, int M, int N);
+
+template <size_t Rows, size_t Cols>
```

Comment:
We're passing in double* in none of the cases where this is called. In fact that is a compile error.
From http://stackoverflow.com/questions/8767166/passing-a-2d-array-to-a-c-function/17569578#17569578
> In C++ passing the array by reference without losing the dimension information is probably the safest, since one needn't worry about the caller passing an incorrect dimension (compiler flags when mismatching). However, this isn't possible with dynamic (freestore) arrays; it works for automatic (usually stack-living) arrays only i.e. the dimensionality should be known at compile time.

This really is giving us more safety and is definitely worth any additional compile time and memory footprint.